src/nativeAssetManager.js

```diff
@@ -160,14 +161,26 @@ export function newNativeAssetManager(win) {
   /*
    * Entry point to search for placeholderes and set up postmessage roundtrip
    * to retrieve native assets. Looks for placeholders for the given adId and
-   * fires a callback after the native html is updated.
+   * fires a callback after the native html is updated. If no placeholders found
+   * and requestAllAssets flag is set in the tag, postmessage roundtrip
+   * to retrieve native assets that have a value on the corresponding bid
    */
   function loadAssets(adId, cb) {
-    const placeholders = scanForPlaceholders(adId);
+    const placeholders = scanForPlaceholders(adId), flag = (typeof win.pbNativeData !== 'undefined');
+
+    if(flag && win.pbNativeData.hasOwnProperty('assetsToReplace')) {
+        win.pbNativeData.assetsToReplace.forEach((asset) => {
+          const key = (asset.match(/hb_native_/i)) ? asset : NATIVE_KEYS[asset];
+          if(key) {placeholders.push(key);}
+        });
+    }
 
     if (placeholders.length > 0) {
       callback = cb;
       requestAssets(adId, placeholders);
+    }else if(flag && win.pbNativeData.hasOwnProperty('requestAllAssets') && win.pbNativeData.requestAllAssets) {
+      callback = cb;
+      requestAllAssets(adId);
```

Comment:
I think we suppose check `requestAllAssets` first. If it is true, then `requestAllAssets`. Only when `requestAllAssets` is not true, then `requestAssets` for  `placeholders`.

Reply:
Can you describe the use case for this? I would think that if placeholders are detected/defined, then requesting all assets should not be a requirement as the needed assets would be known.

Reply:
2 reasons:
1) for Use Case 1 - “Render Replacements”, and Use Case 2 - “Ad Template as String”, these 2 use cases both have placementholder token inside html. pbNativeTagData.requestAllAssets = true; will be useless if the code check placementholders exist first;
2) for Use Case 1 - “Render Replacements”, for html `<img class="pb-icon" alt="icon" height="10" width="10">`, there suppose have some code which get asset of `##hb_native_icon##`, and set src of the image be the value of `##hb_native_icon##`. But inside html, there is no placeholder for `##hb_native_icon##`, so the asset won't get pulled in existing logic.

Reply:
To make it easier, I will create a PR against your branch, so you will understand what I mean.

Reply:
unfortunately my account can't push commit into git@github.com:mmoschovas/prebid-universal-creative.git repo, even inside the my own branch.
Here is the code change I made, I change function fireNativeCallback() to:
```
  function fireNativeCallback(){
    const adElements = findAdElements(AD_ANCHOR_CLASS_NAME);
    for (let i = 0; i < adElements.length; i++) {
      adElements[i].addEventListener('click', function(event) {
        loadClickTrackers(event, window.pbNativeData.adId);
      }, true);
    }
    const imgElements = win.document.getElementsByTagName("img");
    for (let i = 0; i < imgElements.length; i++) {
      if (imgElements[i].dataset && imgElements[i].dataset.src) {
        imgElements[i].src = imgElements[i].dataset.src;
      }
    }
  }
```
with this change, then it will support <img> tag with following format
```
<img data-src="##hb_native_image##" alt="" />
<img data-src="##hb_native_icon##" alt="logo" />
```
This will avoid 400 error if we directly set img.src inside html template.

Reply:
Notice, this is a different solution than using `<img class="pb-icon">`, I think it is better and easier.
And with this solution, it doesn't need `pbNativeTagData.requestAllAssets = true;` for Use Case 1 - “Render Replacements” anymore, since all asset placementholder exists in GAM template html.

Reply:
One more notice, you need to adjust `renderingMocks` inside nativeRenderManager_spec.js to 
```
const renderingMocks = {
  getWindowObject: function() {
    return {
      document: {
        getElementsByTagName: function() {
          return [];
        }
      },
      parent: {
        postMessage: sinon.spy()
      }
    }
  }
};
```